src/settings/FixedDueDateSchedule/FixedDueDateScheduleForm.test.js

```diff
@@ -0,0 +1,512 @@
+    {...rest}
+  >
+    ExpandAllButton
+  </button>
+));
+
+AccordionSet.mockImplementation(({
+  children,
+  onToggle,
+  ...rest
+}) => (
+  // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+  <div
+    data-testid="accordionSet"
+    onClick={() => onToggle({ id: 'generalFixedDueDate' })}
+    {...rest}
+  >
+    {children}
+  </div>
+));
+
+const name = 'test-name';
+
+const messageIds = {
+  about: 'ui-circulation.settings.fDDSform.about',
+  editLabel: 'ui-circulation.settings.fDDS.editLabel',
+  newFixDDSchedule: 'ui-circulation.settings.fDDSform.newFixDDSchedule',
+  uniqueName: 'ui-circulation.settings.fDDS.validate.uniqueName',
+  closeLabel: 'ui-circulation.settings.fDDSform.closeLabel',
+  cancel: 'ui-circulation.settings.fDDSform.cancel',
+  delete: 'stripes-core.button.delete',
+  saveAndClose: 'ui-circulation.settings.common.saveAndClose',
+  schedule: 'ui-circulation.settings.fDDSform.schedule',
+  name: 'ui-circulation.settings.fDDSform.name',
+  description: 'ui-circulation.settings.fDDSform.description',
+  deleteLabel: 'ui-circulation.settings.fDDSform.delete',
+  deleteHeader: 'ui-circulation.settings.fDDSform.deleteHeader',
+  deleteMessage: 'ui-circulation.settings.fDDSform.deleteMessage',
+};
+
+const defaultState = {
+  confirmDelete: false,
+  sections: {
+    generalFixedDueDate: true,
+    schedule: true,
+  },
+};
+
+const okapi = {
+  url: 'test-okapi-url',
+  tenant: 'test-okapi-tenant',
+  token: 'test-okapi-token',
+};
+
+const stripes = {
+  connect: jest.fn(data => data),
+  timezone: 'timezone',
+};
+
+const defaultProps = {
+  initialValues: {},
+  pristine: true,
+  submitting: true,
+  onCancel: jest.fn(),
+  onRemove: jest.fn(),
+  handleSubmit: jest.fn(),
+  stripes,
+  okapi,
+};
+
+describe('FixedDueDateScheduleForm', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Initial render with default props', () => {
+    let container;
+    const fieldCallOrder = {
+      name: 1,
+      description: 2,
+    };
+    const accordionCallOrder = {
+      general: 1,
+      schedule: 2,
+    };
+    const expandAllButtonCallOrder = {
+      render: 1,
+      update: 2,
+    };
+    const buttonCallOrder = {
+      cancel: 1,
+      save: 2,
+    };
+    const accordionSetCallOrder = {
+      render: 1,
+      update: 2,
+    };
+
+    beforeEach(() => {
+      container = render(
+        <FixedDueDateScheduleForm {...defaultProps} />
+      );
+    });
+
+    it('"IconButton" should be rendered with correct props', () => {
+      const expectedProps = {
+        'aria-label': [messageIds.closeLabel],
+        icon: 'times',
+        iconClassName: 'closeIcon',
+        onClick: defaultProps.onCancel,
+      };
+
+      expect(IconButton).toHaveBeenCalledWith(expect.objectContaining(expectedProps), {});
+    });
+
+    it('"onCancel" should be called after clicking on close icon', () => {
```

Comment:
In my opinion this test is useless, because we are cheking inside implementation of  `IconButton` which is mocked.

Reply:
done